File: abstractions/java/lib/src/main/java/com/microsoft/kiota/serialization/SerializationWriterFactory.java
package com.microsoft.kiota.serialization;

import javax.annotation.Nonnull;
/** Defines the contract for a factory that creates SerializationWriter instances. */
public interface SerializationWriterFactory {
    /**
     * Gets the content type this factory creates serialization writers for.
     * @return the content type this factory creates serialization writers for.
     */
    @Nonnull
    String getValidContentType();
    /**
     * Creates a new SerializationWriter instance for the given content type.
     * @param contentType the content type to create a serialization writer for.
     * @return a new SerializationWriter instance for the given content type.
     */
    @Nonnull
    SerializationWriter getSerializationWriter(@Nonnull final String contentType);
}

File: abstractions/php/src/Serialization/SerializationWriterFactory.php
<?php
namespace Microsoft\Kiota\Abstractions\Serialization;

interface SerializationWriterFactory {

    /**
     * Creates a new SerializationWriter instance for the given content type.
     * @param string $contentType the content type to create a serialization writer for.
     * @return SerializationWriter a new SerializationWriter instance for the given content type.
     */
     public function getSerializationWriter(string $contentType): SerializationWriter;

    /**
     * Gets the content type this factory creates serialization writers for.
     * @return string the content type this factory creates serialization writers for.
     */
     public function getValidContentType(): string;
}
